### PR TITLE
add WardsAbstract to dss

### PR DESCRIPTION
**Pull request by iamchrissmith**
Adding a generic Wards Abstract to our DSS interfaces for use in spells

### CODE REVIEW

File: src/dss/WardsAbstract.sol [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.5.12;
+
+interface WardsAbstract {
+    function wards(address) external view returns (uint256);
+    function rely(address) external;
+    function deny(address) external;
+}
```
Comment (naszam) [inline src/dss/WardsAbstract.sol:8]: missing eof newline

**Review by naszam — APPROVED** [review]
LGTM

**Review by naszam — APPROVED** [review]
LGTM

**Review by brianmcmichael — APPROVED** [review]
nice.

**Comment by brianmcmichael** [comment]
FWIW, there is already a dss-exec-lib action available for relying and denying.

https://github.com/makerdao/dss-exec-lib/blob/master/src/DssExecLib.sol#L253-L271

**Comment by iamchrissmith** [comment]
> FWIW, there is already a dss-exec-lib action available for relying and denying.
> 
> https://github.com/makerdao/dss-exec-lib/blob/master/src/DssExecLib.sol#L253-L271

Yeah I was looking to replace/better integrate this code from spells: https://github.com/makerdao/spells-mainnet/blob/master/src/DssSpell.t.base.sol#L26-L28

Do you think I should use the `Authorizable` from that? I feel like this is better to have included in our interfaces repo

**Comment by brianmcmichael** [comment]
Yeah, this is fine here. Your original comment mentioned it was for spells. This will be helpful for tests if we need it.

**Pull request merged**